### PR TITLE
Read method for IssueController

### PR DESCRIPTION
**Pull request by janchk**
#378 
- [x] Read method
- [x] Tests for read method

### CODE REVIEW

**Comment by janchk** [comment]
@malinink check pr pls

File: src/test/java/easytests/api/v1/controllers/IssuesControllerTest.java [hunk 1]
```diff
@@ -129,6 +137,65 @@ public void testListForbidden() throws Exception {
...
+
+        mvc.perform(get("/v1/issues/1")
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNotFound())
+                .andExpect(content().string(""))
+                .andReturn();
+    }
+
+    @Test
+    public void testShowWithQuizesSucces() throws Exception {
```
Comment (malinink) [inline src/test/java/easytests/api/v1/controllers/IssuesControllerTest.java]: there is no need to test that case, it is not differ from the `testShowSuccess` one

File: src/main/java/easytests/api/v1/controllers/IssuesController.java [hunk 2]
```diff
@@ -65,6 +65,22 @@
     /**
      * show(issueId)
      */
+
+    @GetMapping("/{issueId}")
+    public Object show(@PathVariable Integer issueId) throws NotFoundException, ForbiddenException {
+        final IssueModelInterface issueModel = this.getIssueModel(issueId);
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/IssuesController.java]: where is forbidden check?
provide test also

**Review by malinink — CHANGES_REQUESTED** [review]

File: src/main/java/easytests/api/v1/controllers/IssuesController.java (hunk 2 above)
Comment (janchk) [reply src/main/java/easytests/api/v1/controllers/IssuesController.java]: @malinink  Waiting for @miron97 's pr to be accepted to use  ACL for issues. Or should I implement it by myself?

Comment (malinink) [reply src/main/java/easytests/api/v1/controllers/IssuesController.java]: Ok, that Pr is almost done. I will soon merge it.

**Comment by janchk** [comment]
@malinink check changes, pls

File: src/main/java/easytests/api/v1/controllers/IssuesController.java [hunk 3]
```diff
@@ -65,6 +65,26 @@
...
+
+        if (!this.acl.hasAccess(issueModel)) {
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/IssuesController.java]: that will fail, cause there is no subject in issue.

File: src/main/java/easytests/api/v1/controllers/IssuesController.java [hunk 4]
```diff
@@ -65,6 +65,26 @@
...
+            throw new ForbiddenException();
+        }
+        return this.issuesMapper.map(issueModel, Issue.class);
+    }
+
+    private IssueModelInterface getIssueModel(Integer id) throws NotFoundException {
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/IssuesController.java:81]: you need to select model for Auth.

File: src/test/java/easytests/api/v1/controllers/IssuesControllerTest.java [hunk 5]
```diff
@@ -55,6 +61,8 @@
 
     private IssueSupport issueSupport = new IssueSupport();
 
+    private QuizzesSupport quizzesSupport = new QuizzesSupport();
```
Comment (malinink) [inline src/test/java/easytests/api/v1/controllers/IssuesControllerTest.java]: What are the proposals of declaring unused support class?

File: src/test/java/easytests/api/v1/controllers/IssuesControllerTest.java [hunk 6]
```diff
@@ -129,6 +137,57 @@ public void testListForbidden() throws Exception {
     /**
      * show(issueId)
      */
+    @Test
+    public void testShowSuccess() throws Exception {
+        final IssueModelInterface issueModel = new IssueModel();
+        issueModel.map(this.issueSupport.getEntityFixtureMock(0));
+        when(this.issuesOptionsBuilder.forAuth()).thenReturn(new IssuesOptions());
```
Comment (malinink) [inline src/test/java/easytests/api/v1/controllers/IssuesControllerTest.java:140]: you even not use forAuth? but test it =) so funny.

Comment (malinink) [reply src/test/java/easytests/api/v1/controllers/IssuesControllerTest.java:140]: Fix tests after changing fetching issue model logic.
Do not just copy-paste code, think of resons? why this line of code should be here.

**Review by malinink — CHANGES_REQUESTED** [review]

**Comment by janchk** [comment]
@malinink some headache happened yesterday, bugs fixed.

File: src/main/java/easytests/api/v1/controllers/IssuesController.java [hunk 7]
```diff
@@ -65,6 +67,33 @@
...
+        final IssuesOptionsInterface issuesOptionsInterface = this.issuesOptions.forAuth().withSubject(new SubjectsOptions());
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/IssuesController.java]: no need of withSubject here

File: src/test/java/easytests/api/v1/controllers/IssuesControllerTest.java [hunk 8]
```diff
@@ -129,6 +134,57 @@ public void testListForbidden() throws Exception {
...
+
+
+
+        this.mvc.perform(get("/v1/issues/1")
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON))
+                .andExpect(content().json(new JsonSupport()
+                        .with(id, issueModel.getId())
+                        .with(name, issueModel.getName())
```
Comment (malinink) [inline src/test/java/easytests/api/v1/controllers/IssuesControllerTest.java:152]: add subject Identity here

**Review by malinink — CHANGES_REQUESTED** [review]

**Review by malinink — APPROVED** [review]

**Pull request merged**